proxy/coco/src/peer.rs

```diff
@@ -236,7 +230,7 @@ impl Future for Running {
         let err = match self.protocol.poll_unpin(cx) {
             Poll::Ready(val) => match val {
                 Err(e) => Some(Error::Join(e)),
-                Ok(res) => Some(Error::Protocol(res.err().expect("the impossible has happened, someone changed the `!` type in `Running::protocol`, if only there was some sort of function that could show this at compile time"))),
+                Ok(res) => Some(res.err().expect("the impossible has happened, someone changed the `!` type in `Running::protocol`, if only there was some sort of function that could show this at compile time")),
```

Comment:
> if only there was some sort of function that could show this at compile time

Well, it's not too hard to define [into_ok](https://doc.rust-lang.org/std/result/enum.Result.html#method.into_ok) with the type parameters swapped:

```rust
fn into_err<T, E>(r: Result<T, E>) -> E
where
    T: Into<!>
{
    match r {
        Ok(x) => x.into(),
        Err(e) => e,
    }
}
```

Reply:
Ooh that's nice